### DISCUSSION TITLE
Invoking Adguard Home API through the integration

### DISCUSSION THREAD

**Discussion by rcdailey** [hassio-addons/addon-adguard-home#274] [Q&A]
Using NodeRed, I use the HTTP Request node to make a REST API call directly to my adguard home instance. This requires me to set up basic auth headers and a full JSON payload by hand. I'm wondering if it's possible to instead make the API calls through the integration itself, at the very least to take care of authentication for me? I think it would be beneficial to have the Adguard Home authentication centralized in the integration itself, instead of scattered around different NR flows.

If this is possible could you please help me understand how to do it? Thank you.

For more context: I am using API calls instead of the integration directly because I need to be able to block services in client settings (using `POST /clients/update`) instead of blocking them globally. I do this at specific hours each day.

**Comment by frenck** [comment]
I am not sure about your question here?

**Reply by rcdailey** [reply]
What specifically are you not sure about?

**Reply by frenck** [reply]
What you want to change on the addon?

**Reply by rcdailey** [reply]
So I don't want to change the addon per se, I'm more interested in using it as a proxy to make API requests to Adguard Home itself. I apologize if I'm not accurately explaining this.

**Reply by frenck** [reply]
In that case, I don't understand it even less... you don't want to suggest a change for the addon, but want to add a proxy for something? Why proxy anything? The API of adguard is fully exposed?

**Reply by rcdailey** [reply]
I never said I don't want to change the addon, the purpose of this question is to understand if what I want to do is even possible in the first place. I think we're getting ahead of ourselves.

"Proxy" in this case doesn't mean HTTP Proxy, although that might end up being what it actually is. But I mean it in the more general sense: Instead of making an API call directly, I use the addon to hit a specific endpoint in Adguard Home and pass me the response. This would be useful for aspects of Adguard Home that the plugin hasn't been specifically coded for, otherwise I could just do all of this using built in automations in HA.

**Reply by frenck** [reply]
Non of your responses make any sense to me, sorry. 😞
